skrub/_joiner.py

```diff
@@ -45,66 +132,88 @@ class Joiner(TransformerMixin, BaseEstimator):
     suffix : str, default=""
         Suffix to append to the ``aux_table``'s column names. You can use it
         to avoid duplicate column names in the join.
```

Comment:
it is always applied, we decided not to apply it only when there are duplicates (at least for now). note that the pandas and polars approach does not work 100% because they add the suffix only if there are duplicates but then don't check if there are duplicates after adding the suffix. also we thought it is useful to be able to easily know what will be  the output column names. however in a later pr we want to add an option for generating an automatic suffix.

Re what should be the default, `_aux` does make sense although many users will want no suffix (if they don't have duplicated column names), and at the same time `_aux` might be too short to prevent duplicates in some cases.
So I'm not really sure what's best, I guess in many cases users will have to provide their own suffix
WDYT @Vincent-Maladiere and @skrub-data/devs ?

Reply:
Ok thanks, ah yes you are right that checking duplicates after the suffix is (a great asset of the Joiner) changing the logic here. 
I guess this is anyway not a blocking issue for this PR, I'm ok for resolving this with future issues.